debugfs/dump.c

```diff
@@ -122,8 +122,10 @@ static void dump_file(const char *cmdname, ext2_ino_t ino, int fd,
 	}
 	while (1) {
 		retval = ext2fs_file_read(e2_file, buf, blocksize, &got);
-		if (retval)
+		if (retval) {
 			com_err(cmdname, retval, "while reading ext2 file");
+			return;
```

Comment:
if we return we would leak buf, also would need to close the e2_file, so maybe it is better to break instead of return as that would take care of cleanup

Reply:
not sure though how to signal that the file is incomplete

Reply:
Good catch. re signaling, I don't see anything being done to signal that opening, reading, writing, closing the file or allocating memory failed except for a log message. They all just return from the function, and the calling function (`do_dump`) just returns after closing the file descriptor on lines 191-198.

Attempt at e13db13c, let me know what you think.